### PR TITLE
remove connection setup

### PR DESCRIPTION
**Pull request by marten-seemann**
This PR moves the connection setup to go-libp2p-conn, see https://github.com/libp2p/go-libp2p-conn/pull/9.

It should be merged once we activate multi-stream support, as added in https://github.com/libp2p/go-libp2p-transport/pull/20.

### CODE REVIEW

File: muxtest/muxt.go [hunk 1]
```diff
@@ -374,48 +373,53 @@ func SubtestStressNSwarmNConnNStreamNMsg(t *testing.T, tr smux.Transport, nSwarm
...
+	glog.SetOutput(f)
 	tests := []TransportTest{
 		SubtestConstructSwarm,
 		SubtestSimpleWrite,
 		SubtestSimpleWrite100msgs,
 		SubtestStress1Swarm1Conn1Stream1Msg,
 		SubtestStress1Swarm1Conn1Stream100Msg,
 		SubtestStress1Swarm1Conn100Stream100Msg,
-		SubtestStress1Swarm10Conn50Stream50Msg,
+		// SubtestStress1Swarm10Conn50Stream50Msg,
```
Comment (whyrusleeping) [inline muxtest/muxt.go]: is this disabled for some reason?

File: muxtest/muxt.go [hunk 2]
```diff
@@ -374,48 +373,53 @@ func SubtestStressNSwarmNConnNStreamNMsg(t *testing.T, tr smux.Transport, nSwarm
...
-func SubtestStress10Swarm2Conn100Stream100Msg(t *testing.T, tr smux.Transport) {
-	SubtestStressNSwarmNConnNStreamNMsg(t, tr, 10, 2, 100, 100)
+func SubtestStress10Swarm2Conn100Stream100Msg(t *testing.T, lo listenerOpts, streamMuxer smux.Transport) {
+	SubtestStressNSwarmNConnNStreamNMsg(t, lo, streamMuxer, 10, 2, 100, 100)
 }
 
-func SubtestAll(t *testing.T, tr smux.Transport) {
-
+func SubtestAll(t *testing.T, lo listenerOpts, streamMuxer smux.Transport) {
+	f, err := os.Create("./log.txt")
```
Comment (whyrusleeping) [inline muxtest/muxt.go]: remove debug logging code

Comment (marten-seemann) [reply muxtest/muxt.go]: Done.

**Comment by Stebalien** [comment]
Superseded by the refactor

**Pull request closed without merging**